cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/TopicReplicationFactorAnomalyFinder.java

```diff
@@ -75,23 +92,20 @@
         }
       }
     }
+    refreshTopicMinISRCache();
     if (!topicsWithBadReplicationFactor.isEmpty()) {
+      maybeRetrieveAndCacheTopicMinISR(topicsWithBadReplicationFactor);
       Map<Boolean, Set<String>> topicsByFixability = new HashMap<>(2);
-      List<ConfigResource> topicResourcesToCheck = new ArrayList<>(topicsWithBadReplicationFactor.size());
-      topicsWithBadReplicationFactor.forEach(t -> topicResourcesToCheck.add(new ConfigResource(ConfigResource.Type.TOPIC, t)));
-      for (Map.Entry<ConfigResource, KafkaFuture<Config>> entry: _adminClient.describeConfigs(topicResourcesToCheck).values().entrySet()) {
-        try {
-          short topicMinISR = Short.valueOf(entry.getValue().get().get(TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG).value());
-          if (topicMinISR > _targetReplicationFactor) {
+      for (String topic : topicsWithBadReplicationFactor) {
+        if (_cachedTopicMinISR.containsKey(topic)) {
+          short topicMinISR = _cachedTopicMinISR.get(topic).minISR();
+          if (_targetReplicationFactor < topicMinISR + _topicReplicationFactorMargin) {
             topicsByFixability.putIfAbsent(false, new HashSet<>());
-            topicsByFixability.get(false).add(entry.getKey().name());
+            topicsByFixability.get(false).add(topic);
           } else {
             topicsByFixability.putIfAbsent(true, new HashSet<>());
-            topicsByFixability.get(true).add(entry.getKey().name());
+            topicsByFixability.get(true).add(topic);
```

Comment:
Can we move this to a function? -- It seems to have a specific role of checking for minIsr.